app/models/miq_server.rb

```diff
@@ -551,10 +553,20 @@ def logon_status_details
   # Zone and Role methods
   #
   def self.my_guid
-    @@my_guid_cache ||= begin
-      guid_file = Rails.root.join("GUID")
-      File.write(guid_file, SecureRandom.uuid) unless File.exist?(guid_file)
-      File.read(guid_file).strip
+    @my_guid_mutex ||= Mutex.new
```

Comment:
This mutex doesn't seem safe...that is, 2 threads could call my_guid simultaneously and then create 2 separate mutexes.